### PR TITLE
Add Realm.Configuration

### LINKED ISSUE #6081: API documentation for Realm.Configuration

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

This closes #6081 and use [typedoc-plugin-missing-exports](https://www.npmjs.com/package/typedoc-plugin-missing-exports) to resolve missing exports

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

File: packages/realm/src/Configuration.ts [hunk 1]
```diff
@@ -33,20 +33,115 @@ import {
...
+   */
   path?: string;
+  /**
+   * Specifies all the object types in this Realm. **Required** when first creating a Realm at this `path`.
+   * If omitted, the schema will be read from the existing Realm file.
+   * @since 0.10.0
+   */
   schema?: (RealmObjectConstructor<AnyRealmObject> | ObjectSchema)[];
+  /**
+   * **Required** (and must be incremented) after changing the `schema`.
```
Comment (takameyer) [inline packages/realm/src/Configuration.ts:56]: Is that required only on local or also for synced realms?

File: packages/realm/src/Configuration.ts [hunk 2]
```diff
@@ -33,20 +33,115 @@ import {
...
+   * coordinate access to the Realm across threads and processes. If the Realm file is stored in a location
+   * that does not allow the creation of FIFO special files (e.g. FAT32 file systems), then the Realm cannot be opened.
+   * In that case Realm needs a different location to store these files and this property defines that location.
+   * The FIFO special files are very lightweight and the main Realm file will still be stored in the location defined
+   * by the `path` property. This property is ignored if the directory defined by `path` allow FIFO special files.
+   * @since 2.23.0
+   */
   fifoFilesFallbackPath?: string;
   sync?: SyncConfiguration;
   /** @internal */ openSyncedRealmLocally?: true;
```
Comment (takameyer) [inline packages/realm/src/Configuration.ts:87]:
```suggestion
  /** @internal */ 
  openSyncedRealmLocally?: true;
```

**Review by takameyer — APPROVED** [review]
Awesome!  Thanks for doing this! :shipit:

File: packages/realm/src/Configuration.ts (hunk 1 above)
Comment (elle-j) [inline packages/realm/src/Configuration.ts:56]: I first read this quickly as a required field even though the TS type is optional. Perhaps this word rearrangement could help with that:
```suggestion
   * If changing the `schema`, this is **required** and must be incremented.
```

File: packages/realm/src/Configuration.ts [hunk 3]
```diff
@@ -33,20 +33,115 @@ import {
...
+  /**
+   * The function called when opening a Realm for the first time during the life of
+   * a process to determine if it should be compacted before being returned to the user.
+   *
+   * It returns `true` to indicate that an attempt to compact the file should be made. The compaction
+   * will be skipped if another process is accessing it.
+   * @param totalBytes - The total file size (data + free space)
+   * @param usedBytes - The total bytes used by data in the file
+   * @returns `true` if Realm file should be compacted before opening
```
Comment (elle-j) [inline packages/realm/src/Configuration.ts:96]:
```suggestion
   * @param totalBytes - The total file size (data + free space).
   * @param usedBytes - The total bytes used by data in the file.
   * @returns `true` if Realm file should be compacted before opening.
```

File: packages/realm/src/Configuration.ts [hunk 4]
```diff
@@ -33,20 +33,115 @@ import {
...
+   * The function takes two arguments:
+   *   - `oldRealm` - The Realm before migration is performed.
+   *   - `newRealm` - The Realm that uses the latest `schema`, which should be modified as necessary.
+   * @since 0.12.0
+   */
   onMigration?: MigrationCallback;
+  /**
+   * The function called when opening a Realm for the first time. The function can populate the Realm
+   * prior to opening it. When calling the callback, the Realm will be in a write transaction.
+   * @param realm - The newly created Realm
```
Comment (elle-j) [inline packages/realm/src/Configuration.ts:141]:
```suggestion
   * @param realm - The newly created Realm.
```

Comment (elle-j) [inline packages/realm/src/Configuration.ts:141]:
```suggestion
   * @param realm - The newly created Realm.
```

File: packages/realm/src/Realm.ts [hunk 5]
```diff
@@ -231,6 +231,9 @@ type InternalConfig = {
   realmExists?: boolean;
 };
 
+/**
+ * A Realm instance represents a Realm database.
```
Comment (elle-j) [inline packages/realm/src/Realm.ts:235]: Would this be enough?
```suggestion
 * The Realm database.
```

**Review by elle-j — APPROVED** [review]
Great, important to get this in!!

File: packages/realm/src/Configuration.ts (hunk 1 above)
Comment (kneth) [reply packages/realm/src/Configuration.ts:56]: Good point: only for local Realms

**Pull request merged**